org.jacoco.core/src/org/jacoco/core/instr/Instrumenter.java

```diff
@@ -132,13 +124,11 @@ protected String getCommonSuperClass(final String type1,
 	 */
 	public byte[] instrument(final InputStream input, final String name)
 			throws IOException {
-		final byte[] bytes;
 		try {
-			bytes = Java9Support.readFully(input);
+			return instrument(new ClassReader(input));
```

Comment:
If instrumentation fails, then exception won't have location information due to this usage of `instrument(ClassReader)` instead of previously used `instrument(byte[], String)`. Maybe simply preserve `readFully`?

Reply:
This reminds me again recurring old wish to deprecate `instrument(ClassReader)` or at least add a note.

Reply:
@Godin IMHO we can even just remove it. But it's a separate issue though.